cfpb/cfgov-refresh
# Django auth

We want to consolidate the user creation/permission flow, so this is a first attempt at that.  

We now use the Django default view to create a user, where one can set the appropriate permissions. The password that is set does not have to adhere to the password policy, but it also expires immediately, forcing the user to reset their password when they log in (which is what they should be doing anyway). 

One known issue:
- E-mail addresses do not have a unique constraint, so the admin must ensure that they are not creating multiple users with the same e-mail address. If this happens, there'd be weirdness with resetting the password since that uses the e-mail address to reset, and not the username.

To test:
- Pull down this branch, and log in as a superuser 
- From the Django admin, add a user (with a unique email address). Set password to anything of your liking. Make sure to give the user Staff status at the minimum.
- Log out
- Try to log in with that username and password. Notice you are locked out and need to reset your password.
- Reset the password for the email address you provided. Get the reset password link from the console. 
- Try some non-compliant passwords and note that the policy is enforced.
- Try one that is compliant and reset the password successfully.
- Log in with the username and new password

You can also test that the password policy works when trying to change a password as well. 

@kave 
@rosskarchner 
@kurtw
